### PR TITLE
FI-1033: Swap HTTP Gem for rest-client

### PR DESCRIPTION
**Pull request by okeefm**
To solve the ongoing HTTPS issues with bulk-data-server, this PR swaps out the `http` gen in `streamed_ndjson_get` for the `rest-client` gem we use elsewhere. This enables us to use the same SSL verification switching code here as elsewhere.

To test: 
* check out this branch
* build it in dockerland: `docker build -t onchealthit/inferno-program:latest .`
* change to the `inferno-site-overlay` directory
* swap out the production certs for the dev certs in `docker-compose.yml`
* start it in inferno-site-overlay: `docker-compose up --build --force-recreate` 
* visit inferno using your IP address
* Verify that `multi patient` tests fail by default, because of the self-signed cert
* change `disable_verify_peer` to `true` in `inferno-config.yml` in `i-s-o`
* restart docker-compose
* verify that the multi-patient tests pass (or skip, due to data missing)

Pull requests into Inferno require the following items to be completed. Submitter and reviewer 
should check the relevant check boxes when the associated item is done. For items that are not 
applicable, note it's not applicable ("N/A") and check the box. For example, external Pull 
Requests do not require ticket links.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR: FI-1033
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb [hunk 1]
```diff
@@ -344,63 +343,68 @@ def log_and_reraise_if_error(request, response, truncated)
...
+              headers: response.header,
+              body: String.new,
+              truncated: false
+            }
+
+            # Skip processing the last line since the it may not be complete (still appending from stream)
+            next_block = resource_list.pop
+            # Skip if the last_line is empty
+            # Cannot use .blank? since docker-compose complains "invalid byte sequence in US-ASCII" during unit test
+            next_block = String.new next_block unless next_block.nil? || next_block.strip.empty?
```
Comment (Jammjammjamm) [inline lib/modules/onc_program/bulk_data_group_validation_sequence.rb:375]: Does this line actually do anything? To me it looks like it sometimes assigns a copy of `next_block` to itself, and I don't see how that would be useful.

Comment (arscan) [reply lib/modules/onc_program/bulk_data_group_validation_sequence.rb:375]: Is it because Ruby complains now if you mutate strings by default (unless you provide some kind of directive on the top of the file to allow that)?  So we create a new copy of the string?  I recall having to add that kind of stuff here, though I haven't looked too closely to see if this is really related to that.

**Review by yunwwang — APPROVED** [review]
tested using Inferno Bulk data reference server. NDJSON downloads looks good.

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb (hunk 1 above)
Comment (okeefm) [reply lib/modules/onc_program/bulk_data_group_validation_sequence.rb]: I removed it, as discussed in standup last week.

**Review by Jammjammjamm — APPROVED** [review]

**Review by hershilpatel — APPROVED** [review]
Looks good, ran it with disable_verify_peer = true and it worked
